modules/features/search/src/main/java/au/com/shiftyjelly/pocketcasts/search/SearchResultsFragment.kt

```diff
@@ -0,0 +1,158 @@
+package au.com.shiftyjelly.pocketcasts.search
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.fragment.app.viewModels
+import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsSource
+import au.com.shiftyjelly.pocketcasts.compose.AppThemeWithBackground
+import au.com.shiftyjelly.pocketcasts.models.entity.Folder
+import au.com.shiftyjelly.pocketcasts.models.entity.Podcast
+import au.com.shiftyjelly.pocketcasts.models.to.EpisodeItem
+import au.com.shiftyjelly.pocketcasts.models.to.SearchHistoryEntry
+import au.com.shiftyjelly.pocketcasts.models.type.EpisodeViewSource
+import au.com.shiftyjelly.pocketcasts.search.searchhistory.SearchHistoryViewModel
+import au.com.shiftyjelly.pocketcasts.views.fragments.BaseFragment
+import au.com.shiftyjelly.pocketcasts.views.helper.UiUtil
+import dagger.hilt.android.AndroidEntryPoint
+
+private const val ARG_ONLY_SEARCH_REMOTE = "arg_only_search_remote"
+private const val ARG_SOURCE = "arg_source"
+private const val ARG_TYPE = "arg_type"
+@AndroidEntryPoint
+class SearchResultsFragment : BaseFragment() {
+    private val viewModel by viewModels<SearchViewModel>({ requireParentFragment() })
```

Comment:
I used an activity scope view model earlier to share search results and subscribe actions between search (inline results) fragment and search (full-screen) results fragment but noticed that previous search results were shown in the parent search fragment when it was created. 

I now create a full-screen child search results screen and use a parent fragment scope view model to share results + subscribe actions and not mess with the existing implementation. Let me know if you see any drawbacks.

Reply:
Makes sense to me. 👍